client/my-sites/earn/ads/form-settings.jsx

```diff
@@ -17,76 +16,103 @@ import FormTextInput from 'calypso/components/forms/form-text-input';
 import FormTextarea from 'calypso/components/forms/form-textarea';
 import SectionHeader from 'calypso/components/section-header';
 import SupportInfo from 'calypso/components/support-info';
-import { protectForm } from 'calypso/lib/protect-form';
+import { ProtectFormGuard } from 'calypso/lib/protect-form';
+import { useDispatch, useSelector } from 'calypso/state';
 import getSiteUrl from 'calypso/state/selectors/get-site-url';
 import { getWordadsSettings } from 'calypso/state/selectors/get-wordads-settings';
 import isSavingWordadsSettings from 'calypso/state/selectors/is-saving-wordads-settings';
 import { isJetpackSite, getCustomizerUrl } from 'calypso/state/sites/selectors';
 import { getSelectedSite, getSelectedSiteId } from 'calypso/state/ui/selectors';
-import { dismissWordAdsSuccess } from 'calypso/state/wordads/approve/actions';
 import { saveWordadsSettings } from 'calypso/state/wordads/settings/actions';
 
-class AdsFormSettings extends Component {
-	static propTypes = {
-		site: PropTypes.object.isRequired,
-		markChanged: PropTypes.func.isRequired,
-		markSaved: PropTypes.func.isRequired,
-	};
-
-	state = {};
-
-	// @TODO: Please update https://github.com/Automattic/wp-calypso/issues/58453 if you are refactoring away from UNSAFE_* lifecycle methods!
-	UNSAFE_componentWillReceiveProps( { wordadsSettings, site } ) {
-		if ( ( isEmpty( this.state ) && wordadsSettings ) || site?.ID !== this.props.site?.ID ) {
-			this.setState( {
-				...this.defaultSettings(),
+const AdsFormSettings = () => {
+	const translate = useTranslate();
+	const dispatch = useDispatch();
+	const [ settings, setSettings ] = useState( {} );
+	const [ isChanged, setIsChanged ] = useState( false );
+
+	const site = useSelector( ( state ) => getSelectedSite( state ) );
+	const siteId = useSelector( ( state ) => getSelectedSiteId( state ) );
+	const siteUrl = useSelector( ( state ) => getSiteUrl( state, siteId ) );
+	const siteIsJetpack = useSelector( ( state ) => isJetpackSite( state, siteId ) );
+	const isSavingSettings = useSelector( ( state ) => isSavingWordadsSettings( state, siteId ) );
+	const wordadsSettings = useSelector( ( state ) => getWordadsSettings( state, siteId ) );
+	const widgetsUrl = useSelector( ( state ) => getCustomizerUrl( state, siteId, 'widgets' ) );
+
+	const isLoading = ! wordadsSettings || isSavingSettings;
+	const isWordAds = site?.options?.wordads;
+
+	useEffect( () => {
+		// Set default settings if settings are empty
+		const settingsAreEmptyAndWordadsSettingsExist =
+			isEmpty( settings ) && ! isEmpty( wordadsSettings );
+
```

Comment:
The old code checked for `isEmpty( this.state ) && wordadsSettings`.

I've updated the second part to `! isEmpty( wordadsSettings )` to be more precise.